src/sonic-py-common/sonic_py_common/device_info.py

```diff
@@ -582,15 +610,30 @@ def is_smartswitch():
     if not platform:
         return False
 
-    # get platform.json file path
-    platform_json = os.path.join(HOST_DEVICE_PATH, platform, "platform.json")
-    try:
-        with open(platform_json, 'r') as f:
-            platform_cfg = json.loads(f.read())
-            return "DPUS" in platform_cfg
-    except IOError:
+    # Retrieve platform.json data
+    platform_data = get_platform_json_data()
+    if platform_data:
+        return "DPUS" in platform_data
+
+    return False
+
+
+def is_dpu():
```

Comment:
@vvolam can you have something like here? https://github.com/sonic-net/sonic-utilities/blob/master/counterpoll/main.py

Reply:
`switch_type` is take from the config DB. It might not be available at the early boot stage. It is better to rely on `DPU` attribute from the `platform.json` since it is static configuration that is always available